src/containers/AdyenContainer/AdyenContainer.tsx

```diff
@@ -0,0 +1,127 @@
+import type { CoreOptions } from '@adyen/adyen-web/dist/types/core/types';
+import { useCallback, useEffect, useState } from 'react';
+import type DropinElement from '@adyen/adyen-web/dist/types/components/Dropin/Dropin';
+import { useNavigate } from 'react-router';
+
+import useClientIntegration from '../../hooks/useClientIntegration';
+import Adyen from '../../components/Adyen/Adyen';
+import useQueryParam from '../../hooks/useQueryParam';
+import { createAdyenPaymentSession, finalizeAdyenPayment, initialAdyenPayment } from '../../stores/CheckoutController';
+import type { AdyenPaymentSession } from '../../../types/checkout';
+import { reloadActiveSubscription } from '../../stores/AccountController';
+import LoadingOverlay from '../../components/LoadingOverlay/LoadingOverlay';
+import { addQueryParams } from '../../utils/formatting';
+
+import { ADYEN_LIVE_CLIENT_KEY, ADYEN_TEST_CLIENT_KEY } from '#src/config';
+
+type Props = {
+  setUpdatingOrder: (loading: boolean) => void;
+  setPaymentError: (errorMessage?: string) => void;
+  type: AdyenPaymentMethodType;
+  paymentSuccessUrl: string;
+  orderId?: number;
+};
+
+export default function AdyenContainer({ setUpdatingOrder, type, setPaymentError, paymentSuccessUrl, orderId }: Props) {
+  const [processing, setProcessing] = useState(false);
+  const [session, setSession] = useState<AdyenPaymentSession>();
+
+  const { sandbox } = useClientIntegration();
+  const navigate = useNavigate();
+
+  const redirectResult = useQueryParam('redirectResult');
+  const checkPayment = !!redirectResult;
+
+  useEffect(() => {
+    if (checkPayment) return;
+
+    const createSession = async () => {
+      setUpdatingOrder(true);
+
+      try {
+        const session = await createAdyenPaymentSession(window.origin);
+
+        setSession(session);
+      } catch (error: unknown) {
+        if (error instanceof Error) {
+          setPaymentError(error.message);
+        }
+      }
+
+      setUpdatingOrder(false);
+    };
+
+    createSession();
+  }, [setUpdatingOrder, checkPayment, setPaymentError]);
+
+  const onSubmit = useCallback(
+    async (state: AdyenEventData, handleAction: DropinElement['handleAction']) => {
+      if (!state.isValid) return;
+
+      try {
+        setUpdatingOrder(true);
+        setPaymentError(undefined);
+
+        if (orderId === undefined) throw new Error('Order is unknown');
```

Comment:
Throwing an error in a try-catch can be prevented (some consider this to be an antipattern):

```suggestion
        if (orderId === undefined) {
          setPaymentError('Order is unknown');
          return;
        }
```